Cython/Compiler/Symtab.py

```diff
@@ -1362,7 +1374,7 @@ def find_module(self, module_name, pos, relative_level=-1):
             # explicit relative cimport
             # error of going beyond top-level is handled in cimport node
             relative_to = self
-            while relative_level > 0 and relative_to:
+            while relative_level > 0 and relative_to and self._is_package_scope_or_module():
```

Comment:
This will probably be reworked, but it seems wasteful to do something as expensive as this as part of a loop, even though it's obvious that the result will never change. Which, BTW, seems problematic. My guess is that you'd want to check parent packages as well, not just the package directory itself.

Reply:
Also reworked in next version of PR.